tailcfg/derpmap.go

```diff
@@ -14,6 +14,10 @@ type DERPMap struct {
 	// The numbers are not necessarily contiguous.
 	Regions map[int]*DERPRegion
 
+	// DERPHomeHint is a hint to this client to use the provided DERP
+	// region ID as the home region with higher affinity than normal.
+	DERPHomeHint int `json:",omitempty"`
```

Comment:
We should pick a method that allows us to send it to the node as part of the initial connection, to avoid flaps where the node starts, picks a close DERP, and then gets hinted to one that's also close but not the fastest. How about putting this in `MapResponse`?

Reply:
I want to making MapResponse less of a dumping ground than it already is. It's making control too hard to work on.

Reply:
I think the hint should be part of the initial response for the /derpmap because we can generate it there.

But maybe there's a way to do that efficiently? Worst case an HTTP response header.

Reply:
Maybe we define at this capver that DERPMap can be more incremental, with just setting hints (a pointer, not an int) to a new DERPHomeHints type

Reply:
We can deliver a good hint before you have a derp home connection and we should do that. Maybe another endpoint? Or is that what you mean?

Reply:
Concrete proposal below